test/test_unary_ufuncs.py

```diff
@@ -1179,6 +1179,38 @@ def test_i0_special(self, device, dtype):
         t = torch.tensor([inf, -inf, nan], device=device, dtype=dtype)
         self.assertTrue(torch.i0(t).isnan().all())
 
+    @dtypesIfCUDA(*torch.testing.get_all_fp_dtypes())
+    @dtypes(torch.bfloat16, torch.float32, torch.float64)
+    @unittest.skipIf(not TEST_SCIPY, "SciPy not found")
+    def test_special_i0e_vs_scipy(self, device, dtype):
```

Comment:
Would you elaborate on why this bespoke test is needed instead of using reference_numerics?

Reply:
It was mostly motivated by wanting to test
https://github.com/pytorch/pytorch/blob/6da53678c669b19ab8ea7d8f327984e9da822387/test/test_unary_ufuncs.py#L1208-L1212

I was thinking of adding something similar to the extremal case of reference numerics (but for now decided to only test it for this function as it would make more sense to be an independant change)

Reply:
Got it; that sounds good. It would be interesting to see that incorporated into the reference_numerics test.